gears/src/application/node.rs

```diff
@@ -3,60 +3,49 @@ use super::{
     ApplicationInfo,
 };
 use crate::{
-    baseapp::genesis::Genesis,
-    commands::node::{genesis::genesis_account_add, init::init, run::run, AppCommands},
+    baseapp::{genesis::Genesis, QueryRequest, QueryResponse},
+    commands::node::{
+        genesis::genesis_account_add,
+        init::init,
+        run::{run, RouterBuilder},
+        AppCommands,
+    },
     params::Keeper as ParamsKeeper,
 };
 use crate::{
     config::{ApplicationConfig, Config},
     params::ParamsSubspaceKey,
-    rest::RestState,
     types::tx::TxMessage,
 };
-use axum::Router;
 use store_crate::StoreKey;
 
 /// A Gears application.
-pub trait Node: AuxHandler {
+pub trait Node: AuxHandler + RouterBuilder<Self::QReq, Self::QRes> {
```

Comment:
The `QReq` and `QRes` types are not just for the `RouterBuilder` - they're for `BaseApp` to know what types the `typed_query` should expect. The `RouterBuilder` needs to know what they are though since it uses the `typed_query` method.

Reply:
I better try to do it by myself. Hard to explain